### PR TITLE
Feat/poetry

### PR DESCRIPTION
**Pull request by mattreex**
Jira Ticket: [PXP-6244](https://ctds-planx.atlassian.net/browse/PXP-6244)

- replaced requirements.txt, test-requirements.txt and setup.py with pyproject.toml and poetry.lock
- updated Dockerfile to use poetry
- removed outdated TestDockerfile
- updated indexclient to 2.1 and cdislogging to 1.0.0

### Dependency updates
- indexd now uses poetry

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+include = [
+  "NOTICE",
+]
+
+[tool.poetry.dependencies]
+python = "3.6.*"
+authutils = "^4.0.0"
+
+cdislogging = "^1.0.0"
+cdiserrors = {git = "https://github.com/uc-cdis/cdiserrors", rev = "0.1.2"}
```
Comment (paulineribeyre) [inline pyproject.toml:17]: cdiserrors and indexclient are on Pypi

**Review by paulineribeyre — COMMENTED** [review]
awesome!!!
just a few small comments - and please check what lgtm is complaining about, and fix if possible

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+doiclient = {git = "https://github.com/uc-cdis/doiclient", rev = "1.0.0"}
+dosclient = {git = "https://github.com/uc-cdis/dosclient", rev = "1.1.0"}
+gen3authz = "^0.4.0"
+hsclient = {git = "https://github.com/uc-cdis/hsclient", rev = "1.0.0"}
+indexclient = {git = "https://github.com/uc-cdis/indexclient", rev = "1.6.0"}
+jsonschema = "2.5.1"
+flask = "1.1.1"
```
[the remark this answers is not in the document]
Comment (mattreex) [reply pyproject.toml:24]: It was pinned in the requirements.txt. No documentation as to why, so I figured better not to risk it?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+pytest = "^3.0.6"
+pytest-cov = "^2.5.1"
+pytest-flask = "^0.11"
+PyYAML = "5.1"
+responses = "^0.12.0"
+swagger_spec_validator = "2.6.0"
+
+
+[tool.poetry.scripts]
+indexd = 'bin.indexd:main'
```
[the remark this answers is not in the document]
Comment (mattreex) [reply pyproject.toml:43]: Ah yea, I think it's probably better to just remove all together since it's not actually needed.

**Comment by mattreex** [comment]
> awesome!!!
> just a few small comments - and please check what lgtm is complaining about, and fix if possible

:) I think lgtm is just saying the commits have fixed some alerts.

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulineribeyre) [reply pyproject.toml]: let's use `^` for everything and check if unit/integration tests pass

**Comment by paulineribeyre** [comment]
woops, you're right about lgtm.
one more thing, there's a TestDockerfile that's supposed to let you run the tests without having to install indexd and dependencies locally. Please have a look at fixing it so it runs, or delete it if it's too outdated.

**Review by paulineribeyre — APPROVED** [review]
awesome!!

**Pull request merged**